### PR TITLE
fix: postprocessing fmap intendedfor field assignment

### PR DESCRIPTION
**Pull request by fordmcdonald**
fix: populate field map IntendedFor component, resolved sbref protocol naming issue

### CODE REVIEW

File: tests/integration/test_export_and_heudiconv.py [hunk 1]
```diff
@@ -159,3 +161,49 @@ def test_heudiconv():
 
     # cleanup output -- for debugging comment this out
     shutil.rmtree(bids_root_dir, ignore_errors=True)
+
+
+def test_unauthorized_user_exception_handling():
```
Comment (mirestrepo) [inline tests/integration/test_export_and_heudiconv.py:166]: Can you merge the prior PR and then merge `main` so this doesn't show up here?

File: xnat_tools/bids_utils.py [hunk 2]
```diff
@@ -29,33 +29,78 @@ def insert_intended_for_fmap(bids_dir, sub_list):
...
+    # dump the dictionary to the files
+    if len(json_files):
+        acq_tag = get_acquisition_tag(json_files[0].split("_"))
+        for fmap_file in json_files:
+            if acq_tag != get_acquisition_tag(fmap_file.split("_")):
+                _logger.warning(
+                    "Multiple Field Maps With Diferring Acquisition Tags."
+                    "Resolve FMAP IntendedFor Ambiguities."
+                )
+                break
```
Comment (mirestrepo) [inline xnat_tools/bids_utils.py:93]: Do we want to break? Does it make sense that some of the the  json_files get a `data["IntendedFor"]` and others don't? Or should we have a separate loop for validation and only if all valid, then we have a second loop for writing? This is always confusing.. :)

**Review by mirestrepo — CHANGES_REQUESTED** [review]
This is looking great! I added few questions/comments, although minor. The only blocking requirement is that I think we need to add some tests for the post-processing. Maybe I didn't have any before :(, but this is so confusing, that I think we should

File: xnat_tools/bids_utils.py (hunk 2 above)
Comment (fordmcdonald) [reply xnat_tools/bids_utils.py]: @mirestrepo I've updated the processing to inserting the IntendedFor fmap field, with an intuitive insertion if only one field map exists along with scans of the same sequence type (functional or diffusion). Also including the code to handle the protocol naming of sbref derivatives.

File: tests/integration/test_export_and_heudiconv.py (hunk 1 above)
Comment (fordmcdonald) [reply tests/integration/test_export_and_heudiconv.py]: @mirestrepo Can we add a blocking requirement for an updated test under the next feature? I would like to focus on developing a robust test after implementing the IntendedFor acquisition tag key (ace-bold~funcA). Unless you're okay with me completing that work under this feature branch. As for the prior PR / merging to main, I might need to drop by your office for a quick elaboration.

File: xnat_tools/bids_utils.py [hunk 3]
```diff
@@ -82,25 +111,24 @@ def process_fmap_json_files(json_files: list, nii_files: list):
...
-                f.close
-                _logger.info("Done with re-write")
+    if len(fmap_files):
+        if checkFmapAcquistionTags(fmap_files):
+            for fmap in fmap_files:
+                insert_intendedfor_scans(fmap, nii_files)
+
+
+# Verify all acquisition tags of a given list match.
+def checkFmapAcquistionTags(fieldmaps: list):
```
Comment (mirestrepo) [inline xnat_tools/bids_utils.py:121]: The convention for functions names is snake_case

File: xnat_tools/bids_utils.py [hunk 4]
```diff
@@ -30,47 +30,76 @@ def insert_intended_for_fmap(bids_dir, sub_list):
...
 
             # Assemble list of the functional files to add into the intended for field
-            if os.path.exists(func_path):
+            if funcPathExists:
                 func_files = [f"{sess}/func/{file}" for file in os.listdir(func_path)]
                 nii_func_files = [i for i in func_files if i.endswith(".nii.gz")]
                 _logger.info(f"List of func NII files {nii_func_files}")
-                process_fmap_json_files(bold_json_files, nii_func_files)
+                # If there is one field map, with one list of scans, assume correlation and insert.
+                if len(fmap_files) == 1 and funcPathExists ^ diffPathExists:
```
Comment (mirestrepo) [inline xnat_tools/bids_utils.py:63]: Does this now take into consideration the magnitude and phase diff?

**Review by mirestrepo — APPROVED** [review]
Minor change to the functioned that you put in camelCase and we can chat about the magnitudes/phase diff question

**Pull request merged**